pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from "moment";
+
+import cockpit from 'cockpit';
+import { vmId } from "../helpers.js";
+import { CreateSnapshotModal } from "./vmSnapshotsCreateModal.jsx";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { Button, Tooltip } from '@patternfly/react-core';
+import { InfoAltIcon } from '@patternfly/react-icons';
+
+import './vmSnapshotsTab.css';
+
+const _ = cockpit.gettext;
+
+function prettyTime(unixTime) {
+    moment.locale(cockpit.language, {
+        longDateFormat : {
+            LT: "HH:mm:ss",
+            L: "DD/MM/YYYY",
+        }
+    });
+    const yesterday = _("Yesterday");
+    const today = _("Today");
+    moment.locale(cockpit.language, {
+        calendar : {
+            lastDay : `[${yesterday}] LT`,
+            sameDay : `[${today}] LT`,
+            sameElse : "L LT"
+        }
+    });
+
+    return moment(Number(unixTime) * 1000).calendar();
+}
+
+class VmSnapshotsTab extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showCreateSnapshotModal: false,
+        };
+
+        this.openCreateSnapshot = this.openCreateSnapshot.bind(this);
+        this.closeCreateSnapshot = this.closeCreateSnapshot.bind(this);
+    }
+
+    openCreateSnapshot() {
+        this.setState({ showCreateSnapshotModal: true });
+    }
+
+    closeCreateSnapshot() {
+        this.setState({ showCreateSnapshotModal: false });
+    }
+
+    render() {
+        const { vm, dispatch } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Snapshots allow you to revert to an earlier state if something goes wrong")}
```

Comment:
But only if you take them!
If this string is aimed towards people unfamiliar with snapshot functionality, I the the text need to be a little bit more specific. Can't think of a good exact string right now, though.

Reply:
@KKoukiou also had a comment about that line here: https://github.com/cockpit-project/cockpit/pull/13100#discussion_r444889834
Perhaps we can discuss it there.